### PR TITLE
feat: add login with email and password

### PR DESCRIPTION
**Pull request by SwiftXLight**
**Description changes:** 
- Add endpoint for login
- Make validation if email and password match

[link to Jira ticket](https://smartdc23.atlassian.net/browse/SMAR-37?atlOrigin=eyJpIjoiNGJlNzQ4YTg1YTBiNGRlOGFlNzAyMTE2ZjBlYTk0NTciLCJwIjoiaiJ9)

**PR checklist:**
1.1 Common
1.1.1 types for input and output params
1.1.2 no any, should be also added to eslint rules
1.1.3 try/catch for any async function
1.1.4 no magic numbers
1.1.5 compare only with constants not with strings
(instead of user === ‘admin’, better user === roles.admin)
1.1.6 no ternary operator inside ternary operator (bad example: question ? first:
second ? third: fourth)
1.1.7 avoid similar types with duplicating by generics
1.1.8 no console.log in pr
1.1.9 .env file should be in .gitignore
1.1.10 delete commented code if it's not part of documentation
1.1.11 no links in the code, env links should be in env file (for example: server url),
constant links (for example default avatar URL) should be in constant file
1.1.12 constants and function names should be lowercase.
1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant
variable
1.1.14 import rules. imports should come in a specific order: node modules, absolute
path, relative path
1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’,
const EACH_DAY_15h_15min = ‘15 3 * * *’

1.2 Backend
1.2.1 swagger for each endpoint
1.2.2 less requests to db, all data should be taken with one query
1.2.3 `public` in method use only is use function externally
1.2.4 use ConfigService instead of `process.env`

**Successful login:**
image
**Wrong password:**
image
**Wrong email:**
image

### CODE REVIEW

File: src/modules/doctor/doctor.controller.ts [hunk 1]
```diff
@@ -1,18 +1,21 @@
...
+@Controller('create_doctor')
 export default class DoctorController {
   constructor(private readonly doctorService: DoctorService) {}
 
   @ApiOperation({ summary: 'Doctor creation' })
   @ApiResponse({ status: 201, type: Doctor })
   @Post()
-  createDoctor(@Body() doctorDto: CreateDoctorDto): Promise<Doctor> {
-    return this.doctorService.createDoctor(doctorDto);
+  createDoctor(
```
Comment (lyudad) [inline src/modules/doctor/doctor.controller.ts]: this link not protected

File: src/modules/doctor/doctor.service.ts [hunk 2]
```diff
@@ -17,23 +17,36 @@ export default class DoctorService {
...
 
-      return doctor;
+      return newDoctor.generatedMaps[0] as Doctor;
     } catch (err) {
       throw new HttpException(`${err}`, HttpStatus.INTERNAL_SERVER_ERROR);
     }
   }
 
+  async getDoctorByEmail(email: string): Promise<Doctor> {
+    const user = await this.doctorRepository
```
Comment (lyudad) [inline src/modules/doctor/doctor.service.ts]: Screenshot 2023-04-05 at 18 13 14

File: src/modules/doctor/dto/create-doctor.dto.ts [hunk 3]
```diff
@@ -1,6 +1,15 @@
...
+  IsEmail,
+  IsEnum,
+  IsNotEmpty,
+  IsPhoneNumber,
+  IsString,
+  Length,
+  Matches,
+} from 'class-validator';
 import { Role } from 'src/shared/enums';
+import { NAME_MIN_LENGTH, PASSWORD_REGEX } from '../../../shared/consts';
```
Comment (lyudad) [inline src/modules/doctor/dto/create-doctor.dto.ts:12]: ? read git checklist

File: src/modules/doctor/dto/login-doctor.dto.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { ApiProperty } from '@nestjs/swagger';
+import { IsEmail, IsNotEmpty, IsString, Matches } from 'class-validator';
+import { PASSWORD_REGEX } from '../../../shared/consts';
```
Comment (lyudad) [inline src/modules/doctor/dto/login-doctor.dto.ts]: read git checklist

File: src/modules/doctor/doctor.service.ts (hunk 2 above)
Comment (SwiftXLight) [reply src/modules/doctor/doctor.service.ts]: fixed

File: src/modules/doctor/dto/create-doctor.dto.ts (hunk 3 above)
Comment (SwiftXLight) [reply src/modules/doctor/dto/create-doctor.dto.ts:12]: fixed

File: src/modules/doctor/dto/login-doctor.dto.ts (hunk 4 above)
Comment (SwiftXLight) [reply src/modules/doctor/dto/login-doctor.dto.ts]: fixed

File: src/modules/doctor/doctor.controller.ts (hunk 1 above)
Comment (SwiftXLight) [reply src/modules/doctor/doctor.controller.ts]: it was unnecessary controller from setup - deleted

**Review by lyudad — APPROVED** [review]

**Pull request merged**